File: 3639-zero-array-transformation-i/3639-zero-array-transformation-i.cpp
class Solution {
public:
    bool isZeroArray(vector<int>& nums, vector<vector<int>>& queries) {
        int n = nums.size();
        vector<int> nums2(n+1);

        for(int i = 0 ; i < queries.size(); i++) {
            nums2[queries[i][0]]++;
            nums2[queries[i][1] + 1]--;
        }
        for(int i = 1 ; i < n ; i++) {
            nums2[i] += nums2[i - 1];
        }

        for(int i = 0 ; i < n; i++) {
            if(nums[i] > nums2[i]) return false;
        }

        return true;
    }
};

File: 3639-zero-array-transformation-i/3639-zero-array-transformation-i.py
class Solution:
    def isZeroArray(self, nums: List[int], queries: List[List[int]]) -> bool:
        n = len(nums)
        nums2 = [0]*(n + 1)

        for i in queries:
            nums2[i[0]] += 1
            nums2[i[1] + 1] -= 1

        for i in range(1, n):
            nums2[i] += nums2[i - 1]

        for i in nums2: 
            print(i, end = " ")

        for i in range(0, n):
            if nums[i] > nums2[i]: return False;
        
        return True;

        